test/random/TestInverseTransformSamplingCustom.cpp

```diff
@@ -0,0 +1,234 @@
+            double myVal = pow(position(i)[d]-mean, p+1);
+            valL += myVal;
+            },
+            Kokkos::Sum<double>(temp));
+        Kokkos::fence();
+        MPI_Reduce(&temp, &gtemp, 1, MPI_DOUBLE, MPI_SUM, 0, ippl::Comm->getCommunicator());
+        moms[p] = gtemp/(ntotal-1); // Bessel's correction
+    }
+}
+
+void WriteErrorInMoments(double *moms, double *moms_ref, int P){
+    Inform csvout(NULL, "data/error_moments_custom_dist.csv", Inform::APPEND);
+    csvout.precision(10);
+    csvout.setf(std::ios::scientific, std::ios::floatfield);
+    for(int i=0; i<P; i++){
+        csvout << moms_ref[i] << " " << moms[i] << " " << fabs(moms_ref[i] - moms[i]) << endl;
+    }
+    ippl::Comm->barrier();
+}
+
+int main(int argc, char* argv[]) {
+    ippl::initialize(argc, argv);
+    {
+        ippl::Vector<int, 2> nr   = {20, 20};
+        unsigned int ntotal = 1000000;
+
+        ippl::NDIndex<2> domain;
+        for (unsigned i = 0; i < Dim; i++) {
+            domain[i] = ippl::Index(nr[i]);
+        }
+
+        ippl::e_dim_tag decomp[Dim];
+        for (unsigned d = 0; d < Dim; ++d) {
+            decomp[d] = ippl::PARALLEL;
+        }
+
+        ippl::Vector<double, Dim> rmin   = -4.;
+        ippl::Vector<double, Dim> rmax   = 4.;
+        ippl::Vector<double, Dim> length = rmax - rmin;
+
+        ippl::Vector<double, Dim> hr     = length / nr;
+        ippl::Vector<double, Dim> origin = rmin;
+
+        const bool isAllPeriodic = true;
+
+        Mesh_t mesh(domain, hr, origin);
+
+        ippl::FieldLayout<Dim> fl(domain, decomp, isAllPeriodic);
+
+        ippl::detail::RegionLayout<double, Dim, Mesh_t> rlayout(fl, mesh);
+
+        int seed = 42;
+        using size_type = ippl::detail::size_type;
+        GeneratorPool rand_pool64((size_type)(seed + 100 * ippl::Comm->rank()));
+        
+        // example of sampling normal/uniform in one and harmonic in another with custom functors
+        const int DimP = 4; // dimension of parameters in the pdf
+        const double mu = 1.0;
+        const double sd = 0.9;
+        const double parH[DimP] = {mu, sd, 0.5, 2.*pi/(rmax[1]-rmin[1])*4.0}; // paramters of pdf
+        
+        using DistH_t = ippl::random::Distribution<double, Dim, DimP, custom_pdf, custom_cdf, custom_estimate>;
+        using samplingH_t = ippl::random::InverseTransformSampling<double, Dim, Kokkos::DefaultExecutionSpace, DistH_t>;
+
+        DistH_t distH(parH);
+        samplingH_t samplingH(distH, rmax, rmin, rlayout, ntotal);
+        ntotal = samplingH.getLocalNum();
+        view_type positionH("positionH", ntotal);
+        samplingH.generate(positionH, rand_pool64);
+        
+        const int P = 6; // number of moments to check, i.e. E[x^i] for i = 1,...,P
+        double moms1_ref[P];
+        double moms1[P];
+        
+        // compute error in moments of 1st dimension
+        moms1_ref[0] = mu;
+        NormDistCentMoms(sd, P, moms1_ref);
```

Comment:
Please follow the naming conventions for functions.

Reply:
Done.